Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs

```diff
@@ -292,11 +297,30 @@ class ButtonClickListener : Object, IOnClickListener
 
 			#endregion
 
-			public void OnClick(global::Android.Views.View v)
+			public void OnClick(AView v)
 			{
 				var renderer = v.Tag as ButtonRenderer;
 				((IButtonController)renderer?.Element)?.SendClicked();
 			}
 		}
+
+		class ButtonTouchListener : Object, IOnTouchListener
+		{
+			public static readonly Lazy<ButtonTouchListener> Instance = new Lazy<ButtonTouchListener>(() => new ButtonTouchListener());
+
+			public bool OnTouch(AView v, AMotionEvent e)
+			{
+				var renderer = v.Tag as ButtonRenderer;
+				if (e.Action == AMotionEventActions.Down)
+				{
+					((IButtonController)renderer?.Element)?.SendPressed();
+				}
+				else if (e.Action == AMotionEventActions.Up)
+				{
+					((IButtonController)renderer?.Element)?.SendReleased();
```

Comment:
Do you want to use if-else clause like below?

```c#
public bool OnTouch(AView v, AMotionEvent e)
{
	var renderer = v.Tag as ButtonRenderer;
	if (renderer != null)
	{
		var buttonController = (IButtonController)renderer.Element;
		if (e.Action == AMotionEventActions.Down)
		{
			buttonController?.SendPressed();
		}
		else if (e.Action == AMotionEventActions.Up)
		{
			buttonController?.SendReleased();
		}
	}
	return false;
}
```

If right, I'll update it. However, this kind of casting is already use in existing code. 
e.g. ((IButtonController)renderer?.Element)?.SendClicked();